source/Nevermore/Advanced/ThreadSafeAsyncEnumerable.cs

```diff
@@ -9,21 +9,21 @@ namespace Nevermore.Advanced
     public class ThreadSafeAsyncEnumerable<T> : IAsyncEnumerable<T>
     {
         readonly Func<IAsyncEnumerable<T>> innerFunc;
-        readonly SemaphoreSlim semaphore;
+        readonly DeadlockAwareLock deadlockAwareLock;
 
-        public ThreadSafeAsyncEnumerable(IAsyncEnumerable<T> inner, SemaphoreSlim semaphore) : this(() => inner, semaphore)
+        public ThreadSafeAsyncEnumerable(IAsyncEnumerable<T> inner, DeadlockAwareLock deadlockAwareLock) : this(() => inner, deadlockAwareLock)
```

Comment:
This is the actual bug. We were previously passing a `DeadlockAwareLock` here already so there should be no functional difference but we did set a trap for our future selves which we've hopefully now removed.